packages/compass-e2e-tests/helpers/compass.js

```diff
@@ -373,9 +408,22 @@ async function afterTests({ keychain, compass }) {
       compass = null;
     }
   } finally {
-    if (keychain) {
-      keychain.reset();
+    keychain.reset();
+  }
+}
+
+async function printLogs(compass) {
```

Comment:
I looked to see if there's a way to attach something like an event handler for every log line so we can have things fail immediately if an error or unhandled rejection or similar occurs in the browser. I did that once with puppeteer where I had a promise that rejects the moment an error occurs inside the browser. Then every command was wrapped to do Promise.race(browserError, whatever-the-command-is-doing). So you can have the e2e test fail at or at least close to the thing that triggered the error. Which is nicer than waiting for a timeout to expire or the tests just continue on and something else later fails because the state is now mangled and you have to work backwards..

But I can't find a way to do that in webdriver land, so the best we can do for now (I think) is to print browser errors. I suppose we could assert that no errors occurred during the test? Maybe others have better ideas.

Reply:
I am not sure it's worth the time investment tbh, maybe if we move tests to playwright (which is similar to puppeteer) in the future